File: Java/src/test/java/workshops/testingil/dirtytests/exercise/Guru99PlaywrightTest.java
package workshops.testingil.dirtytests.exercise;

import com.microsoft.playwright.*;
import com.microsoft.playwright.options.*;
import org.junit.jupiter.api.Test;

import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;

public class Guru99PlaywrightTest {

    @Test
    public void registerAndCheckFlights() {

        try (Playwright playwright = Playwright.create()) {
            Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setHeadless(true));
            BrowserContext context = browser.newContext();
            Page page = context.newPage();

            page.navigate("https://demo.guru99.com/test/newtours/register.php");
            page.locator("input[name=\"firstName\"]").click();
            page.locator("input[name=\"firstName\"]").fill("Jim");
            page.locator("input[name=\"firstName\"]").press("Tab");
            page.locator("input[name=\"lastName\"]").fill("Smith");
            page.locator("input[name=\"lastName\"]").press("Tab");
            page.locator("input[name=\"phone\"]").press("Tab");
            page.locator("#userName").fill("jimsmith@google.com");
            page.locator("#userName").press("Tab");
            page.locator("input[name=\"address1\"]").click();
            page.locator("input[name=\"address1\"]").fill("19 Hope st.");
            page.locator("input[name=\"address1\"]").press("Tab");
            page.locator("input[name=\"city\"]").fill("New York City");
            page.locator("input[name=\"city\"]").press("Tab");
            page.locator("input[name=\"state\"]").fill("New York");
            page.locator("input[name=\"state\"]").press("Tab");
            page.locator("input[name=\"postalCode\"]").press("Tab");
            page.getByRole(AriaRole.COMBOBOX).selectOption("UNITED STATES");
            page.locator("#email").click();
            page.locator("#email").fill("jimsmith@google.com");
            page.locator("#email").press("Tab");
            page.locator("input[name=\"password\"]").fill("pwd");
            page.locator("input[name=\"password\"]").press("Tab");
            page.locator("input[name=\"confirmPassword\"]").fill("pwd");
            assertThat((page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Submit")))).isEnabled();;
            page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Submit")).click();

            assertThat(page.locator("body")).containsText("Dear Jim Smith,");
            assertThat(page.locator("body")).containsText("Note: Your user name is jimsmith@google.com.");
            page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("sign-in")).click();

            page.locator("input[name=\"userName\"]").click();
            page.locator("input[name=\"userName\"]").fill("jimsmith@google.com");
            page.locator("input[name=\"userName\"]").press("Tab");
            page.locator("input[name=\"password\"]").fill("pwd");
            page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Submit")).click();

            assertThat(page.getByRole(AriaRole.CELL, new Page.GetByRoleOptions().setName("Login Successfully").setExact(true))).isVisible();
            assertThat(page.locator("p").filter(new Locator.FilterOptions().setHasText("Thank you for Loggin."))).isVisible();
            page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Flights")).click();

            page.navigate("https://demo.guru99.com/test/newtours/index.php");
            assertThat(page.locator("p:nth-child(4) > img")).isVisible();
            assertThat(page.getByText("Atlanta to Las Vegas")).isVisible();
            assertThat(page.getByText("Boston to San Francisco")).isVisible();
            assertThat(page.getByText("Phoenix to San Francisco")).isVisible();
            assertThat(page.getByText("New York to Chicago")).isVisible();
            assertThat(page.getByText("Los Angeles to Chicago")).isVisible();
            assertThat(page.locator("body")).containsText("$398");
            assertThat(page.locator("body")).containsText("$513");
            assertThat(page.locator("body")).containsText("$168");
            assertThat(page.locator("body")).containsText("$198");
            assertThat(page.locator("body")).containsText("$213");

            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Cruises"))).isVisible();
            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Vacations"))).isVisible();
            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Hotels"))).isVisible();
            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("CONTACT"))).isVisible();
            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("SIGN-ON"))).isVisible();
            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Salon Travel"))).isVisible();

            page.navigate("https://demo.guru99.com/test/newtours/reservation.php");
            page.getByRole(AriaRole.RADIO).nth(1).check();
            page.locator("select[name=\"passCount\"]").selectOption("3");
            page.locator("select[name=\"fromDay\"]").selectOption("16");
            page.locator("select[name=\"fromMonth\"]").selectOption("9");
            page.locator("select[name=\"toPort\"]").selectOption("Paris");
            page.locator("select[name=\"toMonth\"]").selectOption("9");
            page.locator("select[name=\"toDay\"]").selectOption("17");
            page.getByText("Economy class Business class").click();
            page.getByRole(AriaRole.RADIO).nth(3).check();
            page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Submit")).click();

            assertThat(page.locator("body")).containsText("After flight finder - No Seats Avaialble");
            page.navigate("https://demo.guru99.com/test/newtours/index.php");
            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("your destination"))).isVisible();
            assertThat(page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("featured vacation destinations"))).isVisible();
            assertThat(page.locator("form[name=\"home\"]")).containsText("Registered users can sign-in here to find the lowest fare on participating airlines.");
            assertThat(page.locator("body")).containsText("Always carry a travel first aid kit with bandages, antacids, aspirin, bee sting wipes, and other basic necessities.");
        }
    }
}

File: CSharp/DirtyTestsWorkshop/Exercise/Guru99PlaywrightTest.cs
using NUnit.Framework;
using System.Text.RegularExpressions;
using Microsoft.Playwright;
using Microsoft.Playwright.NUnit;

namespace DirtyTestsWorkshop.Exercise
{
    [TestFixture]
    internal class Guru99PlaywrightTest : PageTest
    {
        [Test]
        public async Task registerAndCheckFlights()
        {
            await Page.GotoAsync("https://demo.guru99.com/test/newtours/register.php");
            await Page.Locator("input[name=\"firstName\"]").ClickAsync();
            await Page.Locator("input[name=\"firstName\"]").FillAsync("Jim");
            await Page.Locator("input[name=\"firstName\"]").PressAsync("Tab");
            await Page.Locator("input[name=\"lastName\"]").FillAsync("Smith");
            await Page.Locator("input[name=\"lastName\"]").PressAsync("Tab");
            await Page.Locator("input[name=\"phone\"]").PressAsync("Tab");
            await Page.Locator("#userName").FillAsync("jimsmith@google.com");
            await Page.Locator("#userName").PressAsync("Tab");
            await Page.Locator("input[name=\"address1\"]").ClickAsync();
            await Page.Locator("input[name=\"address1\"]").FillAsync("19 Hope st.");
            await Page.Locator("input[name=\"address1\"]").PressAsync("Tab");
            await Page.Locator("input[name=\"city\"]").FillAsync("New York City");
            await Page.Locator("input[name=\"city\"]").PressAsync("Tab");
            await Page.Locator("input[name=\"state\"]").FillAsync("New York");
            await Page.Locator("input[name=\"state\"]").PressAsync("Tab");
            await Page.Locator("input[name=\"postalCode\"]").PressAsync("Tab");
            await Page.GetByRole(AriaRole.Combobox).SelectOptionAsync("UNITED STATES");
            await Page.Locator("#email").ClickAsync();
            await Page.Locator("#email").FillAsync("jimsmith@google.com");
            await Page.Locator("#email").PressAsync("Tab");
            await Page.Locator("input[name=\"password\"]").FillAsync("pwd");
            await Page.Locator("input[name=\"password\"]").PressAsync("Tab");
            await Page.Locator("input[name=\"confirmPassword\"]").FillAsync("pwd");
            await Page.GetByRole(AriaRole.Button, new() { Name = "Submit" }).ClickAsync();

            await Expect(Page.Locator("body")).ToContainTextAsync("Dear Jim Smith,");
            await Expect(Page.Locator("body")).ToContainTextAsync("Note: Your user name is jimsmith@google.com.");
            await Page.GetByRole(AriaRole.Link, new() { Name = "sign-in" }).ClickAsync();

            await Page.Locator("input[name=\"userName\"]").ClickAsync();
            await Page.Locator("input[name=\"userName\"]").FillAsync("jimsmith@google.com");
            await Page.Locator("input[name=\"userName\"]").PressAsync("Tab");
            await Page.Locator("input[name=\"password\"]").FillAsync("pwd");
            await Page.GetByRole(AriaRole.Button, new() { Name = "Submit" }).ClickAsync();

            await Expect(Page.GetByRole(AriaRole.Cell, new() { Name = "Login Successfully" }).Nth(1)).ToBeVisibleAsync();
            await Expect(Page.Locator("p").Filter(new() { HasText = "Thank you for Loggin." })).ToBeVisibleAsync();
            await Page.GetByRole(AriaRole.Link, new() { Name = "Flights" }).ClickAsync();

            await Page.GotoAsync("https://demo.guru99.com/test/newtours/index.php");
            await Expect(Page.Locator("p:nth-child(4) > img")).ToBeVisibleAsync();
            await Expect(Page.GetByText("Atlanta to Las Vegas")).ToBeVisibleAsync();
            await Expect(Page.GetByText("Boston to San Francisco")).ToBeVisibleAsync();
            await Expect(Page.GetByText("Phoenix to San Francisco")).ToBeVisibleAsync();
            await Expect(Page.GetByText("New York to Chicago")).ToBeVisibleAsync();
            await Expect(Page.GetByText("Los Angeles to Chicago")).ToBeVisibleAsync();
            await Expect(Page.Locator("body")).ToContainTextAsync("$398");
            await Expect(Page.Locator("body")).ToContainTextAsync("$513");
            await Expect(Page.Locator("body")).ToContainTextAsync("$168");
            await Expect(Page.Locator("body")).ToContainTextAsync("$198");
            await Expect(Page.Locator("body")).ToContainTextAsync("$213");

            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "Cruises" })).ToBeVisibleAsync();
            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "Vacations" })).ToBeVisibleAsync();
            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "Hotels" })).ToBeVisibleAsync();
            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "CONTACT" })).ToBeVisibleAsync();
            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "SIGN-ON" })).ToBeVisibleAsync();
            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "Salon Travel" })).ToBeVisibleAsync();

            await Page.GotoAsync("https://demo.guru99.com/test/newtours/reservation.php");
            await Page.GetByRole(AriaRole.Radio).Nth(1).CheckAsync();
            await Page.Locator("select[name=\"passCount\"]").SelectOptionAsync("3");
            await Page.Locator("select[name=\"fromDay\"]").SelectOptionAsync("16");
            await Page.Locator("select[name=\"fromMonth\"]").SelectOptionAsync("9");
            await Page.Locator("select[name=\"toPort\"]").SelectOptionAsync("Paris");
            await Page.Locator("select[name=\"toMonth\"]").SelectOptionAsync("9");
            await Page.Locator("select[name=\"toDay\"]").SelectOptionAsync("17");
            await Page.GetByText("Economy class Business class").ClickAsync();
            await Page.GetByRole(AriaRole.Radio).Nth(3).CheckAsync();
            await Page.GetByRole(AriaRole.Button, new() { Name = "Submit" }).ClickAsync();

            await Expect(Page.Locator("body")).ToContainTextAsync("After flight finder - No Seats Avaialble");
            await Page.GotoAsync("https://demo.guru99.com/test/newtours/index.php");
            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "your destination" })).ToBeVisibleAsync();
            await Expect(Page.GetByRole(AriaRole.Link, new() { Name = "featured vacation destinations" })).ToBeVisibleAsync();
            await Expect(Page.Locator("form[name=\"home\"]")).ToContainTextAsync("Registered users can sign-in here");
            await Expect(Page.Locator("body")).ToContainTextAsync("Always carry a travel first aid kit");

        }
    }
}
